ui/imports/shared/views/AssetsViewNew.qml

```diff
@@ -0,0 +1,331 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import QtQml.Models 2.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Popups.Dialog 0.1
+
+import AppLayouts.Wallet.controls 1.0
+import shared.controls 1.0
+import shared.popups 1.0
+import utils 1.0
+
+import SortFilterProxyModel 0.2
+
+
+Control {
+    id: root
+
+    /**
+      Expected model structure:
+
+        key         [string] - unique identifier of a token, e.g "0x3234235"
+        symbol      [string] - token's symbol e.g. "ETH" or "SNT"
+        name        [string] - token's name e.g. "Ether" or "Dai"
+        icon        [url]    - token's icon url
+        balance     [double] - tokens balance is the commonly used unit, e.g. 1.2 for 1.2 ETH, used
+                               for sorting and computing market value
+        balanceText [string] - formatted and localized balance. This is not done internally because
+                               it may depend on many external factors
+        error       [string] - error message related to balance
+
+        marketDetailsAvailable [bool]   - specifies if market datails are available for given token
+        marketDetailsLoading   [bool]   - specifies if market datails are available for given token
+        marketPrice            [double] - specifies market price in currently used currency
+        marketChangePct24hour  [double] - percentage price change in last 24 hours, e.g. 0.5 for 0.5% of price change
+
+        communityId   [string] - for community assets, unique identifier of a community, e.g. "0x6734235"
+        communityName [string] - for community assets, name of a community e.g. "Crypto Kitties"
+        communityIcon [url]    - for community assets, community's icon url
+
+        position    [int]  - if custom order available, display position defined by the user via token management
+        canBeHidden [bool] - specifies if given token can be hidden (e.g. ETH should be always visible)
+    **/
+    property var model
+
+    // enables global loading state useful when real data are not yet available
+    property bool loading
+
+    // shows/hides list sorter
+    property bool sorterVisible
+
+    // allows/disables choosing custom sort order from a sorter
+    property bool customOrderAvailable
+
+    // switches configuring right click menu
+    property bool sendEnabled: true
+    property bool swapEnabled: true
+    property bool swapVisible: true
+
+    // global market data error, presented for all tokens expecting market data
+    property string marketDataError
+
+    // formatting function for fiat currency values
+    property var formatFiat: balance => `${balance.toLocaleString(Qt.locale())} USD`
```

Comment:
This is just default value that user can override. The user knows the currency it's using so the function can take just a value. User is responsible for adding any decorations like symbol, it can be done easily form outside using `LocaleUtils` but not from inside - I would have to add LocaleUtils parameters to the API unnecessarily like currency symbol.

I could use the placeholder like "FIXME" here but I think it's nice to have the component with reasonable default values and therefore working "out of the box".

Reply:
`LocaleUtils` is a singleton, you can call it from anywhere; and for the current currency symbol... I'm thinking this should also be part of the singleton/LocaleUtils; it's really global/unique for the whole app

Reply:
> This is just default value that user can override.

Yeah that's the weak point, ppl will not correctly override unless they are forced to

Reply:
Ok, so to have it looking reasonably formatted by default but being obvious that requires customization I can change `USD` -> `XYZ`. Would it be ok?

Reply:
You might as well just display the "FIXME" string there so that it's obvious :)